code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,291 @@
+	keybind_signal = COMSIG_XENOABILITY_LEASH_BALL
+	/// List of all our spiderlings
+	var/list/mob/living/carbon/xenomorph/spiderling/spiderlings
+	/// Max amount of spiderligns
+	var/max_spiderlings = 5
+
+/datum/action/xeno_action/create_spiderling/New(Target)
+	. = ..()
+	spiderlings = list()
+
+/// The action to create spiderlings
+/datum/action/xeno_action/create_spiderling/action_activate()
+	. = ..()
+	if(spiderlings.len >= max_spiderlings)
+		owner.visible_message(span_notice("We have reached the maximum amount of spiderlings"))
+		return fail_activate()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(!do_after(X, 0.5 SECONDS, TRUE, X, BUSY_ICON_DANGER))
+		return fail_activate()
+	/// This creates and stores the spiderling so we can reassign the owner for spider swarm and cap how many spiderlings you can have at once
+	var/new_spiderling = new /mob/living/carbon/xenomorph/spiderling(owner.loc, owner, owner)
+	add_spiderling(new_spiderling)
+	succeed_activate()
+	add_cooldown()
+
+/datum/action/xeno_action/create_spiderling/proc/add_spiderling(mob/living/carbon/xenomorph/spiderling/new_spiderling)
+	RegisterSignal(new_spiderling, COMSIG_MOB_DEATH, .proc/remove_spiderling)
+	spiderlings += new_spiderling
+
+/datum/action/xeno_action/create_spiderling/proc/remove_spiderling(datum/source)
+	SIGNAL_HANDLER
+	spiderlings -= source
+	var/datum/action/xeno_action/spider_swarm/spider_swarm_action = owner.actions_by_path[/datum/action/xeno_action/spider_swarm]
+	if(!spider_swarm_action)
+		return
+	if(source != spider_swarm_action.current_controlling_spiderling)
+		return
+	var/next_spiderling
+	if(spiderlings.len >= 1)
+		next_spiderling = pick(spiderlings)
+	else if(spiderlings.len <= 0)
+		next_spiderling = null
+	if(!next_spiderling)
+		spider_swarm_action.switch_to_mother()
+		return
+	spider_swarm_action.switch_to_next_spiderling(next_spiderling)
+
+// ***************************************
+// *********** Spider Swarm
+// ***************************************
+/datum/action/xeno_action/spider_swarm
+	name = "spider_swarm" // change to in character name later
+	ability_name = "spider_swarm" // change to in character name later
+	mechanics_text = " Turn into a swarm of spiderlings "
+	action_icon_state = "ravage" // temporary until I get my own icons
+	plasma_cost = 1
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_SPIDER_SWARM
+	/// The spiderling we are controlling right now
+	var/mob/living/carbon/xenomorph/spiderling/current_controlling_spiderling
+	/// how many spiderlings should spawn to replace widow
+	var/amount_of_spiderlings = 3
+
+/datum/action/xeno_action/spider_swarm/action_activate()
+	. = ..()
+	if(!do_after(owner, 3 SECONDS, TRUE, owner, BUSY_ICON_DANGER))
+		return fail_activate()
+	/// We store the spiderling that we want to posses in order to make the other spiderlings follow it
+	current_controlling_spiderling = new /mob/living/carbon/xenomorph/spiderling(get_turf(owner), owner)
```

Comment:
you dont need the entire typepath since

var/datum/car/C
C = new
C == a car instance